### DISCUSSION TITLE
How to get `BodyManager` from `PhysicsSystem` to call `GetActiveBodiesUnsafe()`

### DISCUSSION THREAD

**Discussion by AndreaCatania** [jrouwe/JoltPhysics#586] [Q&A] [answered]
Hello, Just before the `PhysicsSystem::update` I would like to fetch the list of active bodies IDs using `BodyManager::GetActiveBodiesUnsafe()` and then fetch the body pointer using `BodyManager::GetBody()`.
On my end it's safe to do so because I would lock the BodyManager during this operation.

However, in order to do that I need to access `BodyManager` but I didn't find any way to do it from `PhysicsSystem`; can you please point me to how to do that?

Thanks :pray:

**Comment by jrouwe** [comment]
The BodyManager is an internal class and cannot be accessed directly. PhysicsSystem::GetActiveBodies was created to access the active body IDs in a safe way.

**Reply by AndreaCatania** [reply]
Since I need to run the simulation in a single thread, it would be safe for me to use `GetActiveBodiesUnsafe` and copy the vector is not really that useful. I wonder, do you think it make sense to expose `GetActiveBodiesUnsafe` API, so that I can read the array without copying it?

**Reply by jrouwe** [reply, marked as answer]
Done in ed4e4d4d90388fc2e0b8e0d351b3965ac0c1d412

**Reply by AndreaCatania** [reply]
> Done in ed4e4d4d90388fc2e0b8e0d351b3965ac0c1d412

Lovely thx a bunch!
